File: erubit.platform/src/main/java/a/erubit/platform/course/Progress.java
package a.erubit.platform.course;

import android.content.res.Resources;

import a.erubit.platform.android.App;
import a.erubit.platform.R;

public abstract class Progress {
    long nextInteractionDate = 0;
    public long interactionDate = 0;
    public long trainDate = 0;

    public int familiarity = 0;

    public int progress = 0;

    public String getExplanation() {
        Resources r = App.getContext().getResources();
        if (interactionDate == 0)
            return r.getString(R.string.unopened);

        return r.getString(
                R.string.progress_explanation,
                progress, familiarity);
    }
}

File: erubit.platform/src/main/java/a/erubit/platform/course/Progress.kt
package a.erubit.platform.course

import a.erubit.platform.R
import android.content.Context


abstract class Progress {
	var nextInteractionDate: Long = 0
	var interactionDate: Long = 0
	var trainDate: Long = 0
	var familiarity = 0
	var progress = 0

	open fun getExplanation(context: Context): String {
		val r = context.resources
		return when (interactionDate) {
			0L -> r.getString(R.string.unopened)
			else -> r.getString(
				R.string.progress_explanation,
				progress, familiarity)
		}
	}
}
